flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/DefaultContext.java

```diff
@@ -55,15 +58,31 @@ public DefaultContext(Configuration flinkConfig, List<CustomCommandLine> command
                 flinkConfig, PluginUtils.createPluginManagerFromRootFolder(flinkConfig));
 
         Options commandLineOptions = collectCommandLineOptions(commandLines);
+
+        final List<URL> dependencies = new ArrayList<>();
+        // add python dependencies by default
+        try {
+            URL location =
+                    Class.forName(
+                                    "org.apache.flink.python.PythonFunctionRunner",
+                                    false,
+                                    Thread.currentThread().getContextClassLoader())
+                            .getProtectionDomain()
+                            .getCodeSource()
+                            .getLocation();
+            if (Paths.get(location.toURI()).toFile().isFile()) {
+                dependencies.add(location);
+            }
+        } catch (URISyntaxException | ClassNotFoundException e) {
+            throw new SqlExecutionException("Failed to find flink-python jar.", e);
```

Comment:
throw SqlGatewayException. In the `SqlGateway#startSqlGateway`, we will notice users it's a bug if the type of the exception is not SqlGatewayException.

Reply:
Make sense.